src/main/java/by/exadel/internship/service/InternshipService.java

```diff
@@ -33,12 +33,59 @@ public GuestInternshipDTO getById(UUID uuid) {
     }
 
     public List<GuestInternshipDTO> getAll() {
-        MDC.put("className", InternshipService.class.getSimpleName());
+        putClassNameInMDC();
         log.info("Try to get all Internships");
         List<Internship> internships = internshipRepository.findAll();
         List<GuestInternshipDTO> guestInternshipDTOList = internShipMapper.map(internships);
-        log.info("Successfully list of Internships");
+        log.info("Successfully got list of Internships");
         return guestInternshipDTOList;
 
     }
+
+    public List<GuestInternshipDTO> getAllDeleted() {
+        putClassNameInMDC();
+        log.info("Try to get deleted Internships");
+        List<Internship> deletedInternships = internshipRepository.findAllDeleted();
+        List<GuestInternshipDTO> guestDeletedInternshipDTOList = internShipMapper.map(deletedInternships);
+        log.info("Successfully got list of deleted Internships");
+        return guestDeletedInternshipDTOList;
+    }
+
+    public GuestInternshipDTO getDeletedInternshipById(UUID uuid) {
+        putClassNameInMDC();
+        log.info("Try to get deleted Internships with uuid= " + uuid);
+        Internship deletedInternship = internshipRepository.findDeletedById(uuid);
+        if (deletedInternship == null) {
+            throw new NotFoundException("No such deleted Internship with uuid = " + uuid +
+                    " in DB", "uuid.invalid");
+        }
+        log.info("Successfully got deleted Internships with uuid= " + uuid);
+        GuestInternshipDTO guestDeletedInternshipDTO = internShipMapper.toGuestInternshipDTO(deletedInternship);
+        return guestDeletedInternshipDTO;
+    }
+
+    public GuestInternshipDTO doActiveDeletedInternshipById(UUID uuid) {
+        putClassNameInMDC();
+        log.info("Try to return deleted Internship with uuid= " + uuid + " to List if Internships");
+        internshipRepository.updateDeletedById(uuid);
```

Comment:
Should be these two operations transactional?

Reply:
Done